source/github-events/on-newsfeed-load.ts

```diff
@@ -1,17 +1,17 @@
 import select from 'select-dom';
 
-export default async function onNewsfeedLoad(callback: VoidFunction): Promise<void> {
-	const observer = new MutationObserver((([{addedNodes}]) => {
-		callback();
+import {createFragmentLoadListener} from './on-fragment-load';
 
-		// If the newly-loaded fragments allows further loading, observe them
-		for (const node of addedNodes) {
-			if (node instanceof Element && select.exists('.ajax-pagination-form', node)) {
-				observer.observe(node, {childList: true});
-			}
-		}
-	}));
+function onNewsfeedPageLoad(event: Event, callback: EventListener, signal: AbortSignal): void {
+	select('.news .ajax-pagination-form[action^="/dashboard-feed"]')?.addEventListener('page:loaded', event => { // Older newsfeed events are paginated through a <form> element
+		onNewsfeedPageLoad(event, callback, signal);
+	}, {signal});
 
-	// Start from the main container
-	observer.observe(select('.news')!, {childList: true});
+	callback(event);
+}
+
+export default function onNewsfeedLoad(callback: EventListener, signal: AbortSignal): void {
+	createFragmentLoadListener('.news include-fragment[src="/dashboard-feed"]', event => {
+		onNewsfeedPageLoad(event, callback, signal);
+	}, signal);
 }
```

Comment:
Rewrote the broken `onNewsfeedLoad` listener (only used by `show-names`) to use GitHub events instead of a `MutationObserver`.

Reply:
Was this broken too?

We should use the new selector observer wherever possible and move away from other listeners, especially when nested like this one.

But if this already fixes an issue and it works, it's fine

Reply:
> Was this broken too?

Yes, since GitHub changed the DOM a lot when introducing the "For you" tab, and now there's a lot more nested elements that are progressively loaded and don't have very specific classes/attributes to target.

This could also have been fixed by adding `subtree: true` to the listener, but listening to events and running the feature only when needed is probably a lot more efficient.

Reply:
> This could also have been fixed by adding `subtree: true` to the listener

See #5886. We don't have to think about that anymore 🎉 

The new observer is not strictly equivalent however, for example the elements must be "not hidden" or else the observer can't see them.